torch/lib/THD/base/ChannelUtils.cpp

```diff
@@ -181,28 +181,29 @@ int connect(const std::string& address, port_type port, bool wait, int timeout)
       }
 
       socklen_t err_len = sizeof(errno);
-      SYSCHECK(getsockopt(socket, SOL_SOCKET, SO_ERROR, &errno, &err_len));
+      ::getsockopt(socket, SOL_SOCKET, SO_ERROR, &errno, &err_len);
+      /* `errno` is set when:
+       *   1. `getsockopt` has failed
+       *   2. there was awaiting error in socket (we save error on `errno` variable)
```

Comment:
What is actually not right here? Also I don't understand what are you trying to say in the proposal you gave.